GRDB/Record/PersistableRecord+Encodable.swift

```diff
@@ -40,7 +42,21 @@ private struct PersistableRecordKeyedEncodingContainer<Key: CodingKey> : KeyedEn
             // This allows us to encode Date as String, for example.
             encode((value as! DatabaseValueConvertible), key.stringValue)
         } else {
-            try value.encode(to: PersistableRecordEncoder(codingPath: [key], encode: encode))
+            do {
+                 try value.encode(to: PersistableRecordEncoder(codingPath: [key], encode: encode))
+            } catch {
+                // If value.encode does not work e.g. "unkeyed encoding is not supported" then see if model can be stored as JSON
+                let encodeError = error
+                do {
+                    let json = try JSONEncoder().encode(value)
+                    guard let modelAsString = String(data: json, encoding: .utf8) else {
+                        throw JsonStringError.covertStringError("Error, could not make string out of JSON data")
+                    }
+                    return encode(modelAsString, key.stringValue)
+                } catch {
+                    fatalError("Encode error: \(encodeError), tried to encode to Json, got error: \(error)")
```

Comment:
We should throw, and not fatalError here.

I understand that we hesitate between rethrowing `encodeError` or `error`. What do you think?